gremlin-core/src/test/java/org/apache/tinkerpop/gremlin/process/traversal/strategy/optimization/PathRetractionStrategyTest.java

```diff
@@ -191,6 +206,9 @@ public void doTest() {
                 {__.V().as("a").optional(bothE().dedup().as("b")).
                         choose(select("b"), select("a","b"), project("a").by(select("a"))),
                         "[[[a, b]], [[a, b]], [[a, b]], [[[a, b]]], [[a, b]]]", null},
+                {__.V().as("a").out().program(labeledPathVertexProgram).select("a"), "[]", null},
+                {__.V().as("a").out().program(pathVertexProgram).select("a"), "[]", null},
+                {__.V().as("a").out().program(emptyRequirementsVertexProgram).select("a"), "[[]]", null}
```

Comment:
I don't understand how the assertions work or rather what exactly is being tested here, but having `select('a')` as the last step, would keep the label anyways, so that the VP would have access to it, no matter what its requirements are, right?